aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/DeploymentSteps.java

```diff
@@ -191,9 +192,10 @@ private CommandInput getCliDeploymentCommand(List<List<String>> componentNames,
         for (Map.Entry<String, ComponentDeploymentSpecification> entry : components.entrySet()) {
             String componentName = entry.getKey();
             commandArgs.add(String.format("--merge %s=%s", componentName, entry.getValue().componentVersion()));
-            String updateConfigArgs = getCliUpdateConfigArgs(componentName, configuration);
-            if (!updateConfigArgs.isEmpty()) {
-                commandArgs.add("--update-config '" + updateConfigArgs + "'");
+            String configurationUpdate = getCliUpdateConfigArgs(componentName, configuration);
+            if (!configurationUpdate.isEmpty()) {
+                Files.write(configFilePath, configurationUpdate.getBytes(StandardCharsets.UTF_8));
```

Comment:
What will happen if this file already exists? Is that even possible?

Reply:
It is possible. It will overwrite the existing file and that's the behavior we want too.